File: cpp_msg_srv/src/prsn_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "custom_interfaces/msg/person_status.hpp"

class StatusPublisher : public rclcpp::Node 
{
public:
    StatusPublisher() : Node("string_publisher") 
    {
        publisher = this->create_publisher<custom_interfaces::msg::PersonStatus>("person_status", 10);
        timer = this->create_wall_timer(std::chrono::milliseconds(500),
                                         std::bind(&StatusPublisher::publishStatusCB, this));
        RCLCPP_INFO(this->get_logger(), "Status publisher started!");
    }

private:

    void publishStatusCB()
    {
        auto msg = custom_interfaces::msg::PersonStatus();
        msg.name = std::string("Sajin Sathananthan");
        msg.available = true;
        msg.salary = 12000;
        publisher->publish(msg);
    }

    rclcpp::Publisher<custom_interfaces::msg::PersonStatus>::SharedPtr publisher;
    rclcpp::TimerBase::SharedPtr timer;


};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<StatusPublisher>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: py_msg_srv/py_msg_srv/prsn_status_publisher.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

from custom_interfaces.msg import PersonStatus

class StatusPublisherNode(Node):
    def __init__(self):

        super().__init__('prsn_status_publisher')
        self.get_logger().info('Publisher started!')
        self.pub = self.create_publisher(PersonStatus,"/person_status",10)
        self.create_timer(0.5,self.send_prsn_status)

    def send_prsn_status(self):
        # self.get_logger().info("Hello World! "+str(self.counter))
        msg = PersonStatus()
        msg.name = "Sajin Sathananthan"
        msg.available = True
        msg.salary = 12000
        self.pub.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node=StatusPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__=='__main__':
    main()